tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -415,32 +443,81 @@ func applyReviewsRoutingRules(t *testing.T) {
 	time.Sleep(30 * time.Second)
 }
 
-func visitProductPage(retries int) error {
-	standby := 0
-	for i := 0; i <= retries; i++ {
-		time.Sleep(time.Duration(standby) * time.Second)
-		http.DefaultClient.Timeout = 1 * time.Minute
-		resp, err := http.Get(fmt.Sprintf("%s/productpage", tc.gateway))
+// checkProductPageDirect
+func checkProductPageDirect() {
+	glog.Info("checkProductPageDirect")
+	dumpURL("http://localhost:"+productPagePort+"/productpage", false)
+}
+
+// dumpMixerMetrics fetch metrics directly from mixer and dump them
+func dumpMixerMetrics() {
+	glog.Info("dumpMixerMetrics")
+	dumpURL("http://localhost:"+mixerMetricsPort+"/metrics", true)
+}
+
+func dumpURL(url string, dumpContents bool) {
+	clnt := &http.Client{
+		Timeout: 1 * time.Minute,
+	}
+	status, contents, err := get(clnt, url)
+	glog.Infof("%s ==> %d, error=<%v>", url, status, err)
+	if dumpContents {
+		glog.Infoln(contents)
+	}
+}
+
+func get(clnt *http.Client, url string, headerkv ...string) (status int, contents string, err error) {
+	var req *http.Request
+	req, err = http.NewRequest("GET", url, nil)
+	if err != nil {
+		return 0, "", err
+	}
+
+	for i := 0; i < len(headerkv)/2; i++ {
```

Comment:
might want to log an error or panic if headerkv isn't even (caller bug/error)

Reply:
headers are optional, it is not an error.

Reply:
0 is even
but if someone passes 1 argument it will get silently ignored which is surely not what you intend ? (or 3 or any 2N+1)